glean-core/csharp/Glean/Metrics/StringListMetricType.cs

```diff
@@ -0,0 +1,172 @@
+            }
+
+            Dispatchers.LaunchAPI(() => {
+                SetSync(value);
+            });
+        }
+
+        /// <summary>
+        /// Sets a string list to one or more metric stores in a synchronous way.
+        /// This is only to be used for the glean-ac to glean-core data migration.
+        /// </summary>
+        /// <param name="value">This is a user defined string list.</param>
+        public void SetSync(string[] value)
+        {
+            if (disabled)
+            {
+                return;
+            }
+
+            LibGleanFFI.glean_string_list_set(
+                        this.handle,
+                        value,
+                        value.Length);
+        }
+
+        /// <summary>
+        /// Tests whether a value is stored for the metric for testing purposes only.This function will
+        /// attempt to await the last task(if any) writing to the the metric's storage engine before
+        /// returning a value.
+        /// </summary>
+        /// <param name="pingName"> represents the name of the ping to retrieve the metric for.
+        /// Defaults to the first value in `sendInPings`.</param>
+        /// <returns>true if metric value exists, otherwise false</returns>
+        public bool TestHasValue(string pingName = null)
+        {
+            Dispatchers.AssertInTestingMode();
+
+            string ping = pingName ?? sendInPings[0];
+            return LibGleanFFI.glean_string_list_test_has_value(this.handle, ping) != 0;
+        }
+
+        /// <summary>
+        /// Returns the stored value for testing purposes only.This function will attempt to await the
+        /// last task(if any) writing to the the metric's storage engine before returning a value.
+        /// @throws [NullPointerException] if no value is stored
+        /// </summary>
+        /// <param name="pingName">represents the name of the ping to retrieve the metric for.
+        /// Defaults to the first value in `sendInPings`.</param>
+        /// <returns>value of the stored metric</returns>
+        /// <exception cref="System.NullReferenceException">Thrown when the metric contains no value</exception>
+        public string[] TestGetValue(string pingName = null)
+        {
+            Dispatchers.AssertInTestingMode();
+
+            if (!TestHasValue(pingName))
+            {
+                throw new NullReferenceException();
+            }
+
+            string ping = pingName ?? sendInPings[0];
+          
+            JsonDocument jsonPayload = JsonDocument.Parse(
+                LibGleanFFI.glean_string_list_test_get_value_as_json_string(this.handle, ping).AsString()
+            );
+            JsonElement root = jsonPayload.RootElement;
+            int size = root.GetArrayLength();
+            string[] result = new string[size];
+ 
+            for (int i = 0; i < size; i++)
+            {
+                result[i] = root[i].ToString();
+            }
+            return result;
```

Comment:
Did you try if `return JsoSerializer.Deserialize<string[]>(root);` works?

Reply:
Good idea! It works.